### PR TITLE
Add additional rtw area config

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*

This PR extends the rich-text-widget to allow for custom options to be passed through the per area configuration. This will allow for new Tiptap extensions to pass their own configuration options. A tutorial on this subject and three new improvements to the rich-text-widget will be released to take advantage of this. This change does not alter the existing configuration behavior.

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [X] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/rich-text-widget/ui/apos/components/AposRichTextWidgetEditor.vue [hunk 1]
```diff
@@ -546,7 +546,8 @@ export default {
       return (apos.tiptapExtensions || [])
         .map(extension => extension({
           styles: this.editorOptions.styles.map(this.localizeStyle),
-          types: this.tiptapTypes
+          types: this.tiptapTypes,
```
Comment (haroun) [inline modules/@apostrophecms/rich-text-widget/ui/apos/components/AposRichTextWidgetEditor.vue:549]: I think you can achieve the same goal with 
```javascript
      aposTiptapExtensions() {
        return (apos.tiptapExtensions || [])
          .map(extension => extension({
            ...this.editorOptions,
            styles: this.editorOptions.styles.map(this.localizeStyle),
            types: this.tiptapTypes
          }));
      },
```

**Review by haroun — CHANGES_REQUESTED** [review]

**Review by haroun — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ### Adds
 
 * Add `publicApiCheckAsync` wrapper method (and use it internally) to allow for async permission checks of REST APIs. This feature doesn't introduce any breaking changes.
+* Add ability to add custom rich-text-widget options per area.
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply CHANGELOG.md]: Changed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
+
+* Add ability for custom tiptap extensions to access the options passed to rich text widgets at the area level
+
 ## 3.55.0
 
 ### Adds
 
-* Add `publicApiCheckAsync` wrapper method (and use it internally) to allow for async permission checks of REST APIs. This feature doesn't introduce any breaking changes.
-* Add ability to add custom rich-text-widget options per area.
+* Add `publicApiCheckAsync` wrapper method (and use it internally) to allow for overrides to do async permission checks of REST APIs. This feature doesn't introduce any breaking changes because the default implementation still invokes `publicApiCheck` in case developers have overridden it.
```
Comment (boutell) [inline CHANGELOG.md:13]: Why is this in this PR? 🤔

**Review by boutell — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (BoDonkey) [reply CHANGELOG.md:13]: The PR was opened before the last release date so there ended up being a merge conflict in the CHANGELOG.

**Pull request merged**